src/models/states/fanspeed.ts

```diff
@@ -0,0 +1,7 @@
+import { State } from './state';
+
+export type availableFanSpeeds = 'off' | 'low' | 'medium' | 'high';
+
+export interface FanSpeedState extends State {
+     availableFanSpeeds: availableFanSpeeds;
```

Comment:
Shouldn't `availableFanSpeeds` be an array? I would also let the node-red plugin to define other values as well.